tests/functional/test_install_extras.py

```diff
@@ -104,6 +104,21 @@ def test_nonexistent_options_listed_in_order(script, data):
     assert matches == ['nonexistent', 'nope']
 
 
+def test_install_deprecated_extra(script, data):
+    """
+    Warn about deprecated order of specifiers and extras.
+    """
+    simple = script.site_packages / 'simple'
+
+    result = script.pip(
+        'install', '--no-index', '--find-links=' + data.find_links,
+        'requires_simple_extra>=0.1[extra]', expect_stderr=True,
```

Comment:
Put this in a requirements.txt file and install from it.

We're hitting a bug in the test pipeline in Windows.